tree.go

```diff
@@ -1254,22 +1168,36 @@ func (n *LeafNode) GetProofItems(keys keylist) (*ProofElements, []byte, [][]byte
 	return pe, esses, poass
 }
 
+// Serialize serializes a LeafNode.
+// The format is: <nodeType><stem><bitlist><c1comm><c2comm><children...>
 func (n *LeafNode) Serialize() ([]byte, error) {
-	var bitlist [32]byte
-	children := make([]byte, 0, NodeWidth*32)
+	// Empty value in LeafNode used for padding.
+	var emptyValue [LeafValueSize]byte
+
+	// Create bitlist and store in children LeafValueSize (padded) values.
+	children := make([]byte, 0, NodeWidth*LeafValueSize)
+	var bitlist [bitlistSize]byte
 	for i, v := range n.values {
 		if v != nil {
 			setBit(bitlist[:], i)
 			children = append(children, v...)
-			if len(v) < 32 {
-				padding := make([]byte, 32-len(v))
+			if padding := emptyValue[:32-len(v)]; len(padding) != 0 {
 				children = append(children, padding...)
```

Comment:
Every padding reuses the `emptyValue` and that's fine. `append` is copying always zero-bytes.